### PR TITLE
[SYCL] Add test for sycl::is_compatible supports reqd_work_group_size

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
Impl: https://github.com/intel/llvm/pull/8056

### CODE REVIEW

File: SYCL/OptionalKernelFeatures/is_compatible.cpp [hunk 1]
```diff
@@ -42,5 +46,31 @@ int main() {
...
   }
 
+  if (sycl::is_compatible<GoodWGSize>(Dev)) {
+    Q.submit(
+        [&](sycl::handler &h) { 
+          h.parallel_for<class GoodWGSize>(sycl::range<2>(4, 2),
+                   [=](sycl::item<2> it) [[sycl::reqd_work_group_size(SIZE, SIZE)]] {});
+        });
+    Q.wait();
+    Compatible &= (Dev.get_info<sycl::info::device::max_work_group_size>() > (SIZE + SIZE));
```
Comment (AlexeySachkov) [inline SYCL/OptionalKernelFeatures/is_compatible.cpp]:
```suggestion
    Compatible &= (Dev.get_info<sycl::info::device::max_work_group_size>() > (SIZE * SIZE));
```

File: SYCL/OptionalKernelFeatures/is_compatible.cpp [hunk 2]
```diff
@@ -42,5 +46,31 @@ int main() {
...
+                   [=](sycl::item<1> it) [[sycl::reqd_work_group_size(INT_MAX)]] {});
+        });
+    Q.wait();
+    Compatible &= false;
+    Called = false;
+  }
+  else {
+    Compatible &= true;
+    Called = true;
+  }
```
Comment (AlexeySachkov) [inline SYCL/OptionalKernelFeatures/is_compatible.cpp]: I would rather check if `INT_MAX` exceeds `device::max_work_group_size` and that's true, then assert that `is_compatible<WrongReqWGSize>` returns `false`

**Comment by dm-vodopyanov** [comment]
@intel/llvm-reviewers-runtime , @cperkinsintel , friendly ping

**Comment by dm-vodopyanov** [comment]
@intel/llvm-reviewers-runtime , can you please review this?

**Review by steffenlarsen — APPROVED** [review]
LGTM! Could you please make clang-format happy? 😄

**Comment by dm-vodopyanov** [comment]
> LGTM! Could you please make clang-format happy? 😄

Oh yes, sure, fixed.

**Pull request merged**